### PR TITLE
(#138) Allow to sort search results by timestamps

### PR DESCRIPTION
**Pull request by excelsior**
- Allows to sort search results by the `created_at` or `updated_at` timestamps.
- Sets the default sort order to `updated_at DESC`.

### CODE REVIEW

**Comment by leolara** [comment]
@science @jbaird123 @cwd-mparsons @siuc-nate  I know I am not very present, but in my oponion we should do a push to the Auth-auto-signing and then attack the search API. Unless, have priorities changed?

**Comment by leolara** [comment]
@science @jbaird123 @cwd-mparsons @siuc-nate Ignore my comment, I spoke with Joe about this

**Comment by leolara** [comment]
@excelsior I approved it, but perhaps good idea if @andersoncardoso can look at the code?

File: app/services/search.rb [hunk 1]
```diff
@@ -110,6 +117,12 @@ def search_resource_fields
       end
     end
 
+    def sort_results
+      sort_by = sort_columns.include?(@sort_by) ? @sort_by : 'updated_at'
+      sort_order = %w[asc desc].include?(@sort_order) ? @sort_order : 'desc'
+      @query.order!(sort_by => sort_order)
```
Comment (andersoncardoso) [inline app/services/search.rb]: I guess we can avoid generating arrays on every call to check keys:
```
sort_by = @sort_params[:by].presence || 'updated_at' 
sort_order = @sort_params[:order].presence || 'desc'
@query.order!(sort_by => sort_order)
```

File: spec/services/search_spec.rb [hunk 2]
```diff
@@ -147,4 +147,125 @@
       /processed_resource \@\> '{ \"publisher\": { \"name\": \"someone\" } }'/
     )
   end
+
+  context 'sorting' do
```
Comment (andersoncardoso) [inline spec/services/search_spec.rb]: 👍

File: app/services/search.rb (hunk 1 above)
Comment (excelsior) [reply app/services/search.rb]: Hey Anderson,

Thanks for taking a look. The inclusion check was added in order to defend against non-supported values, so we don't end up with an invalid SQL statement. This surely could be implemented at the Grape level instead, but I'd rather have all the search related stuff in one place, i.e. in this service.

**Pull request merged**